src/components/HexInput.tsx

```diff
@@ -1,10 +1,16 @@
 import React, { useState, useEffect, useCallback } from "react";
-import validHex from "../utils/validHex";
+
+import { validHex } from "../utils/validate";
 
 // Escapes all non-hexadecimal characters including "#"
-const escape = (hex) => hex.replace(/([^0-9A-F]+)/gi, "");
+const escape = (hex: string) => hex.replace(/([^0-9A-F]+)/gi, "");
+
+interface Props {
+  color: string;
+  onChange: (newColor: string) => void;
+}
```

Comment:
`color` and `onChange` are both optional. Should we add `?` mark?

Reply:
I'll use `Partial` instead for type re-usability, but yes, thanks, missed that.

Reply:
Could I ask about advantages of `Partial`?
It feels to me that `?` will it more clear for those who trying to understand how everything works. But I'm not a pro in TS and I probably just doesn't know something)

Reply:
So `?` is for a specific field, while `Partial<object>` is the same as labeling every field with `?`. 

In general it's just a shorthand, but it's nice for being able to reuse the interface. The component is a pretty good example. Take the following:

```
interface Props {
  x: string;
}

const Component = (props: Partial<Props>) => {
  const { x }: Props = { x: "", ...props};
  ...
}
```

In the above example, the component will accept a partial representation of `Props`. So `x` is completely optional. We can then reuse that interface in the component to describe our de-structured props to ensure we're not forgetting a field while setting default values.

Had we used `?` in `Props` then `x`, even after being destructured, would actually be defined as `string | undefined`.

Using `Partial` gets around having two interfaces; one for the optional props, and one to tell TS that you know your destructured values are not `undefined`.